lib/Router.js

```diff
@@ -0,0 +1,198 @@
+						}
+						else {
+							conversations[numbers[i].number] = [ null, null, null ];
+						}
+					}
+					resolve();
+				};
+			});
+		});
+		return promise;
+	};
+
+	router.use(bodyParser.json());
+	router.use("/", express.static("public"));
+
+	/*
+	*iterates through the list of PhoneNumber objects to find the one with the matching
+	*'number' field.
+	*/
+	this.getNumber = function (number) {
+		for (var i = 0; i < this.myNumbers.length; i += 1){
+			if (this.myNumbers[i].number === number){
+				return this.myNumbers[i];
+			}
+		}
+		return null;
+	}
+
+	/*
+	*Catapult post message to search and return a list of numbers matching city and state
+	*/
+	this.searchNumbers = function (city, state) {
+		var promise = new Promise(function (resolve,reject) {
+			catapult.AvailableNumber.searchLocal(client, { state : state, city : city }, function (err, numbers) {
+				if (err) {
+					reject(err);
+				}
+				else {
+					resolve(numbers);
+				}
+			});
+		});
+		return promise;
+	}
+
+	/*
+	*Catapult call to buy a phone number and set the applicationId to this app
+	*/
+	this.orderNumber = function (number) {
+		var promise = new Promise(function (resolve, reject) {
+			catapult.PhoneNumber.create(client, { number : number, applicationId : applicationId }, function (err, phoneNumber) {
+				if (err) {
+					reject(err);
+				}
+				else {
+					resolve(phoneNumber);
+				}
+			});
+		});
+		return promise;
+	}
+
+	this.sendTextFromAlias = function (alias, number, message) {
+		var promise = new Promise(function (resolve, reject) {
+			catapult.Message.create(client, { from : alias, to : number, text : message }, function (err, message) {
+				if (err) {
+					reject(err);
+				}
+				else {
+					resolve();
+				}
+			});
+		});
+		return promise;
+	}
+
+	/*
+	*POSTs to /numbers come from the web interface and contain the fields
+	*'text', 'oneLiner', 'city', and 'state'. On receiving a POST to /numbers, the app searches
+	*for a new number in the city and state specified, orders it, and sends the client
+	*a text from the alias number with instructions.
+	*/
+	router.post("/numbers", function (req, res) {
+		var clientNum = req.body.text;
+		var city = req.body.city;
+		var state = req.body.state;
+		self.searchNumbers(city, state).then(function (numbers) {
+			return self.orderNumber(numbers[0].number);
+		}).then(function (number) {
+			var alias = number.number;
+			conversations[alias] = [ clientNum, null, req.body.oneLiner ];
+			self.myNumbers.push(number);
+			self.getNumber(alias).update({
+				name : JSON.stringify(conversations[alias]), applicationId : applicationId }, function (err, result) {});
+			var firstMsg = "Thank you for choosing TextMe. Reply with your contact's number (ex. \"+15555555555\"). " +
+			"Send \"Get lost\" to erase this number";
+			return self.sendTextFromAlias(alias, clientNum, firstMsg);
+		}).then(function () {
+			res.send();//return 200ok so a timeout doesn't result in multiple requests
```

Comment:
Should this return 200, or 201? Technically you are sort of creating something new, although that new thing does not have a resource URL that it can be requested at, so I could go either way on this one.